skfda/tests/test_stats_std.py

```diff
@@ -0,0 +1,209 @@
+"""Test stats functions."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+import pytest
+
+from skfda import FDataBasis, FDataGrid
+from skfda.exploratory.stats import std
+from skfda.representation.basis import (
+    Basis,
+    BSplineBasis,
+    FourierBasis,
+    MonomialBasis,
+    TensorBasis,
+    VectorValuedBasis,
+)
+
+# Fixtures for test_std_fdatabasis_vector_valued_basis
+
+
+@pytest.fixture(params=[3, 5])
+def vv_n_basis1(request: Any) -> int:
+    """n_basis for 1st coordinate of vector valued basis."""
+    return request.param  # type: ignore[no-any-return]
+
+
+@pytest.fixture(params=[3])
+def vv_n_basis2(request: Any) -> int:
+    """n_basis for 2nd coordinate of vector valued basis."""
+    return request.param  # type: ignore[no-any-return]
+
+
+@pytest.fixture
+def vv_basis1(vv_n_basis1: int) -> Basis:
+    """1-dimensional basis to test for vector valued basis."""
+    # First element of the basis is assumed to be the 1 function
```

Comment:
WDYM? That is not true for the B-spline basis.

Reply:
I should not have used BSpline basis. Fixed.